### PR TITLE
feat: notify user about new version

### LINKED ISSUE #45: Add "New version available" message
When starting the `check` or `run` subcommand of the CLI, we should fetch the latest version in the background. When the user exits the application (on `"close"` event), we should print out a message if there is a newer version available, like gh CLI does. It should look something like this:

```
> deno check main.ts
OK

A new release of deployctl is available: 0.2.0 -> 0.3.0
To upgrade, run `deployctl upgrade`
https://github.com/denoland/deployctl/releases/tag/0.3.0
```

This should only happen if stdin is a tty (`Deno.isatty(Deno.stdin.rid)` returns `true`).

### PR DESCRIPTION
**Pull request by satyarohith**
**How does it work?**
When a user runs deployctl, we use a web worker to fetch and store the latest release information at `$HOME/.deno/deployctl/update.json`. The information is valid for 24 hours. We check if the information is valid on consecutive runs and if the user's version is the latest. If not the latest, we display the update message. And if the information is not valid, we rerun the web worker.

It looks like the below.
```
➜  deployctl check examples/echo.js          
A new release of deployctl is available: 0.2.0 -> 0.3.0
To upgrade, run `deployctl upgrade`
https://github.com/denoland/deployctl/releases/tag/0.3.0

OK
➜  deployctl run examples/echo.js  
A new release of deployctl is available: 0.2.0 -> 0.3.0
To upgrade, run `deployctl upgrade`
https://github.com/denoland/deployctl/releases/tag/0.3.0

Listening on http://0.0.0.0:8080
```

I'm not sure how to write a test for it.

Closes #45

### CODE REVIEW

File: deployctl.ts [hunk 1]
```diff
@@ -56,6 +57,48 @@ const args = parseArgs(Deno.args, {
...
+  // Get the path to the update information json file.
+  const { updatePath } = getConfigPaths();
+  // Try to read the json file.
+  const updateInfoJson = await Deno.readTextFile(updatePath).catch((error) => {
+    if (error.name == "NotFound") return null;
+    console.error(error);
+  });
+  if (updateInfoJson) {
+    const updateInfo = JSON.parse(updateInfoJson) as {
+      lastUpdated: number;
```
Comment (kt3k) [inline deployctl.ts:71]: Maybe `lastFetched`?

**Comment by kt3k** [comment]
The overall flow looks good to me.

> I'm not sure how to write a test for it.

If we extract small utils like `saveConfig`, `restoreConfig`, `isConfigStale` etc, then we can write tests for them and I think those will cover most of the complex part of diff

**Review by kt3k — APPROVED** [review]
Confirmed this works on my machine. Thanks @satyarohith !

LGTM

**Comment by lucacasonato** [comment]
Only just saw this, but requiring `--unstable` now will break for users who previously installed without `--unstable` and are running `deployctl upgrade` now.

**Comment by lucacasonato** [comment]
@satyarohith Unstable errors are uncatchable, so the worker wouldn't fail.

**Pull request merged**